### PR TITLE
Add workflow for updating robottelo image on Quay.

### PR DESCRIPTION
**Pull request by jameerpathan111**
What this does?
This workflow is mainly responsible for updating Robottelo image on Quay container registry.
It also performs code checks before pushing new image to registry.

How this works?
It uses [build-and-push-docker-images](https://github.com/marketplace/actions/build-and-push-docker-images) github action by Docker to build and push robottelo image to Quay.

I have separated codechecks workflow for push and pull_request event.  
It will be triggered whenever there is any change in robottelo repository.
There are two jobs as follows:
1. codechecks
2. robottelo_container: This depends on codechecks job. It will be responsible for pushing new image to Quay.

I have also added a status badge to display build status of workflow.

Test run for this change:
https://github.com/jameerpathan111/robottelo/actions/runs/357673664

Note:
For this to work, we first need to create following [github secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets).
- REGISTRY_SERVER:quay.io
- REGISTRY_USERNAME:satqe-username
- REGISTRY_PASSWORD:satqe-password
- REGISTRY_NAMESPACE:satqe-something

### CODE REVIEW

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
We can keep the Dockerhub setup as-is for now. How about you change this PR to be the one that introduces automated quay builds?

**Comment by jameerpathan111** [comment]
@JacobCallahan made the required changes. Now it will work for Quay.

**Review by JacobCallahan — APPROVED** [review]

**Comment by jameerpathan111** [comment]
@JacobCallahan someone needs to create those secrets before merging this pr.

**Comment by jameerpathan111** [comment]
Do not merge till github secrets are created.

File: .github/workflows/update_robottelo_image.yml
Comment (jyejare) [inline .github/workflows/update_robottelo_image.yml:41]: Do we really need this check for pushing the robottelo image to Quay ?

The code-coverage shouldnt have anything to do with image , I would remove it.

File: .github/workflows/update_robottelo_image.yml
Comment (jyejare) [inline .github/workflows/update_robottelo_image.yml:53]: Same for this as above.

**Review by jyejare — APPROVED** [review]
ACK pending non blocking comments / questions.

File: .github/workflows/update_robottelo_image.yml
Comment (lpramuk) [inline .github/workflows/update_robottelo_image.yml:39]: might be useful? or required? to `export SATELLITE_VERSION=6.9` to avoid ssh checks for version as there is no SAT instance involved. 

robottelo tests are decorated by BZs - the decorator needs to know SAT version for its own logic

**Review by lpramuk — CHANGES_REQUESTED** [review]
Please export SATELLITE_VERSION

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml:41]: @jyejare probably we don't need it. It's just that I have separated the workflow which we currently have, that's why you are seeing this here. If you want I can remove this from here but then should I keep the push event which I removed? 
Currently the main robottelo_container job depends on codechecks job, it's upto us if we want it this way.

@JacobCallahan any thoughts?

File: .github/workflows/update_robottelo_image.yml
Comment (mshriver) [reply .github/workflows/update_robottelo_image.yml:39]: I'd actually like to fix this behavior in robottelo, so that it will default to some version automatically without needing to set this env var. If there's no host to connect to, just pick a version. Downside being we would have to periodically update it (maybe once a year)

File: .github/workflows/update_robottelo_image.yml
Comment (JacobCallahan) [inline .github/workflows/update_robottelo_image.yml:89]:
```suggestion
          registry: ${{ secrets.QUAY_SERVER }}
          username: ${{ secrets.QUAY_USERNAME }}
          password: ${{ secrets.QUAY_PASSWORD }}
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: .github/workflows/update_robottelo_image.yml
Comment (JacobCallahan) [reply .github/workflows/update_robottelo_image.yml:41]: I agree. this is something I was initially thinking as well.

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml]: updated.

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml]: removed

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml]: removed

**Review by JacobCallahan — APPROVED** [review]

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml:39]: Mike, so I shouldn't add it here, right?

Comment (mshriver) [reply .github/workflows/update_robottelo_image.yml:39]: its fine to have it here for now

File: .github/workflows/update_robottelo_image.yml
Comment (mshriver) [inline .github/workflows/update_robottelo_image.yml:3]: Since we accept PRs to mutliple branches, I would be more specific here that this is on master branch only.

I think we're going to want separate image labels for the z-stream branches, and a separate workflow for them.

```
on:
  push:
    branches:
      - master
```

File: README.rst
Comment (mshriver) [inline README.rst:19]: :100:

File: .github/workflows/update_robottelo_image.yml
Comment (mshriver) [reply .github/workflows/update_robottelo_image.yml:3]: Actually I see you've handled this with the git ref setting the image tag, well done.

Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml:3]: yes

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml:40]: Added the SATELLITE_VERSION env variable [here](https://github.com/SatelliteQE/robottelo/pull/8150/files#diff-ee633393f1eca923b9e701f896601dcb1a7a1cd0c55238b8c0bb37c076459898R14).

File: README.rst
Comment (jameerpathan111) [reply README.rst:19]: :)

File: .github/workflows/update_robottelo_image.yml
Comment (mshriver) [inline .github/workflows/update_robottelo_image.yml:38]: Two things that aren't blockers but I think merit discussion.

1. I would not disable pytest warnings.  Currently we see many of those, but I'm working on fixing that, #8163  is part of such effort.  We should not be getting any pytest warnings, but people would have to come here to see them.

2. The output of the workflow is a bit difficult to parse with all of the output being in one expandable block.  Setup-plan is great for debugging when working with pytest and its fixtures, but I think the output is needlessly verbose here. Collection alone satisfies what we want to test and provides a clear list of everything collected.

**Review by mshriver — APPROVED** [review]

File: .github/workflows/update_robottelo_image.yml
Comment (jameerpathan111) [reply .github/workflows/update_robottelo_image.yml:38]: @JacobCallahan @lpramuk @jyejare any thoughts?

**Review by lpramuk — APPROVED** [review]

**Comment by jameerpathan111** [comment]
Thanks everyone :)

**Pull request merged**
